pkg/generate/app/pipeline.go

```diff
@@ -38,20 +39,20 @@ func NewImagePipeline(from string, image *ImageRef) (*Pipeline, error) {
 	}, nil
 }
 
-// NewBuildPipeline creates a new pipeline with components that are
-// expected to be built
+// NewBuildPipeline creates a new pipeline with components that are expected to
+// be built.
 func NewBuildPipeline(from string, input *ImageRef, outputDocker bool, strategy *BuildStrategyRef, env Environment, source *SourceRef) (*Pipeline, error) {
 	name, ok := NameSuggestions{source, input}.SuggestName()
 	if !ok {
-		name = fmt.Sprintf("app%d", rand.Intn(10000))
+		return nil, ErrNameRequired
```

Comment:
newapp_test shouldn't require resources on the server to be real - it
should mock them out.  You should be able to inject a fake Git
implementation.

On Mon, Oct 12, 2015 at 11:49 AM, Rodolfo Carvalho <notifications@github.com
> wrote:

> In pkg/generate/app/pipeline.go
> <https://github.com/openshift/origin/pull/5082#discussion_r41770515>:
>
> >  func NewBuildPipeline(from string, input *ImageRef, outputDocker bool, strategy *BuildStrategyRef, env Environment, source *SourceRef) (*Pipeline, error) {
> >  	name, ok := NameSuggestions{source, input}.SuggestName()
> >  	if !ok {
> > -		name = fmt.Sprintf("app%d", rand.Intn(10000))
> > +		return nil, ErrNameRequired
>
> I would like to add a test case for this in
> https://github.com/openshift/origin/blob/master/pkg/generate/app/cmd/newapp_test.go,
> but haven't found what type of input that will get us into this branch...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/5082/files#r41770515>.
>